superset-frontend/plugins/plugin-chart-echarts/src/BigNumber/BigNumberWithTrendline/transformProps.ts

```diff
@@ -107,13 +108,54 @@ export default function transformProps(
       // sort in time descending order
       .sort((a, b) => (a[0] !== null && b[0] !== null ? b[0] - a[0] : 0));
 
-    bigNumber = sortedData[0][1];
     timestamp = sortedData[0][0];
 
-    if (bigNumber === null) {
-      bigNumberFallback = sortedData.find(d => d[1] !== null);
-      bigNumber = bigNumberFallback ? bigNumberFallback[1] : null;
-      timestamp = bigNumberFallback ? bigNumberFallback[0] : null;
+    const metricValues: number[] = sortedData
+      .map(d => d[1])
+      .filter((value): value is number => value !== null);
+
+    if (metricValues.length > 0) {
+      const sortedValues: number[] = [...metricValues].sort((a, b) => a - b);
+      const totalSum: number = metricValues.reduce(
+        (sum, value) => sum + value,
+        0,
+      );
+      const mid = Math.floor(sortedValues.length / 2);
+
+      switch (aggregation) {
+        case 'SUM':
+          bigNumber = totalSum;
+          break;
+        case 'AVG':
+          bigNumber = totalSum / metricValues.length;
+          break;
+        case 'MIN':
+          bigNumber =
+            metricValues.length > 0 ? Math.min(...metricValues) : null;
+          break;
+        case 'MAX':
+          bigNumber =
+            metricValues.length > 0 ? Math.max(...metricValues) : null;
+          break;
+        case 'MEDIAN':
+          bigNumber =
+            sortedValues.length % 2 === 0
+              ? ((sortedValues[mid - 1] ?? 0) + (sortedValues[mid] ?? 0)) / 2
+              : (sortedValues[mid] ?? 0);
+          break;
+        case 'LAST_VALUE':
+        default:
+          bigNumber = sortedData.length > 0 ? (sortedData[0][1] ?? null) : null;
```

Comment:
You already check that `metricValues` is not empty, so you can just:

```suggestion
          bigNumber = sortedData[0][1];
```